src/pyclts/commands/map.py

```diff
@@ -1,96 +1,121 @@
 """
 Map a given sound inventory list to CLTS
 """
+
 from pyclts.cli_util import add_format, Table
 from pyclts import CLTS
-from csvw.dsv import UnicodeReader
+import csv
+
 
 def register(parser):
-    add_format(parser, default='simple')
-    parser.add_argument(
-        'graphemes',
-        help='the file with the graphemes',
-        )
+    add_format(parser, default="simple")
+    parser.add_argument("graphemes", help="the file with the graphemes")
+
 
 def run(args, test=False):
-    bipa = args.repos.transcriptionsystem('bipa')
-    with UnicodeReader(args.graphemes, delimiter='\t') as reader:
-        data = []
-        for line in reader:
-            data.append(line)
-        header = data[0]
-        data = data[1:]
-    bidx, gidx = header.index('BIPA'), header.index('GRAPHEME')
-    count = 0
-    cluster = set()
-    for i, line in enumerate(data):
-        rg, bg = line[gidx], line[bidx]
-        if bg and bg != '<NA>':
-            if bipa[bg].type != 'unknownsound':
-                pass
+    bipa = args.repos.transcriptionsystem("bipa")
+
+    new_rows = []
+    clusters = set()
+    with open(args.graphemes) as grapheme_file:
+        for row in csv.DictReader(grapheme_file, delimiter="\t"):
+            bipa_grapheme = row["BIPA"]
+            raw_grapheme = row["GRAPHEME"]
+
+            if bipa_grapheme and bipa_grapheme != "<NA>":
+                if bipa[bipa_grapheme].type != "unknownsound":
```

Comment:
As we discussed via email, as an improvement, we want the researcher to know that the bipa-grapheme has been modified, if this is the case, so we would 

1. leave the bipa-grapheme unmodified if `sound.s == row['BIPA']`
2. add a warning, if we modify it: `else: row[BIPA'] = '(!)'+sound.s`